src/nvim/option.c

```diff
@@ -1442,15 +1442,18 @@ do_set (
               goto skip;
             }
 
-            if (adding)
-              value = *(long *)varp + value;
-            if (prepending)
-              value = *(long *)varp * value;
-            if (removing)
-              value = *(long *)varp - value;
-            errmsg = set_num_option(opt_idx, varp, value,
-                errbuf, sizeof(errbuf), opt_flags);
-          } else if (opt_idx >= 0) {                      /* string */
+            if (adding) {
+              value = *(intmax_t *)varp + value;
```

Comment:
It doesn't seem aesthetically better with the tmp variable. I'l leave as is for now unless you have a compelling reason to change it.